posydon/popsyn/binarypopulation.py

```diff
@@ -542,7 +556,7 @@ def combine_saved_files(self, absolute_filepath, file_names, **kwargs):
             tmp_df = pd.DataFrame(
                 index=[self.metallicity],
                 data={'simulated_mass': simulated_mass,
-                      'underlying_mass': initial_total_underlying_mass(df=simulated_mass, **self.kwargs)[0], 
+                      'underlying_mass': initial_total_underlying_mass(df=simulated_mass,df1=simulated_mass_single, df2= simulated_mass_binaries, **self.kwargs)[0], 
```

Comment:
I would suggest adding these as additional data, similar to the simulated mass. 
Also I would  not calculate the underlying mass here, since the underlying mass calculation requires a user-defined input, ie. f_bin_nature. :)
Instead, a new function as part of the population class would be useful. I'm happy to write this after the weekend.